core/types/l1_message_tx.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/common"
+)
+
+const L1MessageTxType = 0x7E
+
+// payload, RLP encoded
+type L1MessageTx struct {
+	Nonce  uint64
+	Gas    uint64          // gas limit
+	To     *common.Address // can not be nil, we do not allow contract creation from L1
```

Comment:
```suggestion
	To     common.Address // can not be nil, we do not allow contract creation from L1
```

Reply:
Let's leave it as a pointer for now, in case we want to support contract creation in the future.